api/room/schema.py

```diff
@@ -68,38 +63,22 @@ class Arguments:
         room_type = graphene.String()
         capacity = graphene.Int(required=True)
         image_url = graphene.String()
-        location_id = graphene.Int()
-        floor_id = graphene.Int(required=True)
+        location_id = graphene.Int(required=True)
         calendar_id = graphene.String()
-        office_id = graphene.Int(required=True)
-        wing_id = graphene.Int()
-        block_id = graphene.Int()
         cancellation_duration = graphene.Int()
         room_tags = graphene.List(graphene.Int)
     room = graphene.Field(Room)
 
     @Auth.user_roles('Admin')
-    def mutate(self, info, office_id, **kwargs):
+    def mutate(self, info, **kwargs):
         validate_empty_fields(**kwargs)
-        verify_ids(office_id, kwargs)
-        get_office = Office.query.filter_by(id=office_id).first()
-        if not get_office:
-            raise GraphQLError("No Office Found")
-
-        admin_roles.create_rooms_update_delete_office(office_id)
+        verify_location_id(kwargs)
+        admin_roles.create_rooms_update_delete_location(kwargs)
         query = Room.get_query(info)
-        exact_query = room_join_office(query)
-        result = exact_query.filter(
-            Office.id == office_id,
-            RoomModel.name == kwargs.get('name'))
-        if result.count() > 0:
+        location = query.filter(
+            func.lower(RoomModel.name) == func.lower(kwargs.get('name')))
```

Comment:
I think for this query you should include only those whose state is active.

Reply:
On it @kampkelly